### PR TITLE
Add NSButton Introspection

### LINKED ISSUE #38: NSButton works
The readme claims that buttons can't be introspected but I tried this and it absolutely works:

```swift
extension View {
    public func introspectButton(customize: @escaping (NSButton) -> ()) -> some View {
        return inject(AppKitIntrospectionView(
            selector: { introspectionView in
                guard let viewHost = Introspect.findViewHost(from: introspectionView) else {
                    return nil
                }
                return Introspect.previousSibling(containing: NSButton.self, from: viewHost)
            },
            customize: customize
        ))
    }
}
```

Example setting button as default action:

```swift
Button(action: {  })
    .introspectButton { button in
        button.keyEquivalent = "\r"
    }
```

Have not tested this against iOS UIButton, just AppKit NSButton. If someone is feeling frisky, they can add proper button support to the library as outlined above.

### PR DESCRIPTION
**Pull request by JannThomas**
This adds NSButton introspection and thereby resolves #38.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@ Changelog
 
 ## master
 
+- Add NSButton Introspection
```
Comment (SplittyDev) [inline CHANGELOG.md:6]: Would be nice to put `.introspectButton` here, since it's in line with other changelog entries.

File: Introspect/ViewExtensions.swift [hunk 2]
```diff
@@ -198,5 +198,10 @@ extension View {
     public func introspectTabView(customize: @escaping (NSTabView) -> ()) -> some View {
         return introspect(selector: TargetViewSelector.siblingContaining, customize: customize)
     }
+    
+    /// Finds a `NSButton` from a `SwiftUI.Button`
+    public func introspectButton(customize: @escaping (NSButton) -> ()) -> some View {
+        return introspect(selector: TargetViewSelector.siblingContaining, customize: customize)
```
Comment (SplittyDev) [inline Introspect/ViewExtensions.swift:204]: I think this should be constrained to `macOS` since it doesn't work on `iOS`.

File: IntrospectTests/UIKitTests.swift [hunk 3]
```diff
@@ -304,6 +304,8 @@ private struct SegmentedControlTestView: View {
 
 @available(iOS 13.0, tvOS 13.0, macOS 10.15.0, *)
 class UIKitTests: XCTestCase {
+    
+
```
Comment (SplittyDev) [inline IntrospectTests/UIKitTests.swift:308]: 💅 unnecessary change

**Review by SplittyDev — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (JannThomas) [reply CHANGELOG.md]: Done

File: Introspect/ViewExtensions.swift (hunk 2 above)
Comment (JannThomas) [reply Introspect/ViewExtensions.swift:204]: There is an #if canImport(AppKit) check around the entire extension and no other method has the change you propose.

File: IntrospectTests/UIKitTests.swift (hunk 3 above)
Comment (JannThomas) [reply IntrospectTests/UIKitTests.swift:308]: Removed.

**Review by SplittyDev — APPROVED** [review]
LGTM 🚀
